internal/debug/flags.go

```diff
@@ -242,3 +281,27 @@ func Exit() {
 		closer.Close()
 	}
 }
+
+func validateLogLocation(logLocation string) error {
+	if err := os.MkdirAll(logLocation, os.ModePerm); err != nil {
+		return fmt.Errorf("error creating the directory: %w", err)
+	}
+
+	if !isWritable(logLocation) {
+		return fmt.Errorf("write access not present for given log location")
+	}
+
+	return nil
+}
+
+func isWritable(path string) bool {
+	// Check if the path is writable by trying to create a temporary file
+	tmpfile := filepath.Join(path, "tmp")
+	f, err := os.Create(tmpfile)
+	if err != nil {
+		return false
+	}
+	f.Close()
+	os.Remove(tmpfile)
+	return true
+}
```

Comment:
Can be simplified somewhat:
```suggestion
func validateLogLocation(path string) error {
	if err := os.MkdirAll(path, os.ModePerm); err != nil {
		return fmt.Errorf("error creating the directory: %w", err)
	}
	// Check if the path is writable by trying to create a temporary file
	tmp := filepath.Join(path, "tmp")
	if f, err := os.Create(tmp); err != nil {
		return err
	} else {
		f.Close()
	}
	return os.Remove(tmp)
}
```